### PR TITLE
feat(beam): Update stable version to 0.4.0.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

We are using a SNAPSHOT version of Beam and should update to a stable version.

**What is the new behavior?**

Beam 0.4.0-TLND with a patch for BEAM-649 and BEAM-1206.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/488/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/488/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/490/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/490/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/493/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/493/)

File: services/components-api-service-rest-all-components/pom.xml [hunk 1]
```diff
@@ -325,11 +325,7 @@
 													</exec>
 												</entryPoint>
 												<runCmds>
-													<run>apk add --no-cache snappy libc6-compat</run>
-													<!--<run>apk -q -U add ca-certificates</run>-->
-													<!--<run>wget -q -O /etc/apk/keys/andyshinn.rsa.pub https://raw.githubusercontent.com/andyshinn/alpine-pkg-glibc/master/andyshinn.rsa.pub</run>-->
-													<!--<run>wget -q https://github.com/andyshinn/alpine-pkg-glibc/releases/download/unreleased/glibc-2.23-r1.apk</run>-->
-													<!--<run>apk add glibc-2.23-r1.apk</run>-->
+													<run>apk add --no-cache libc6-compat</run>
```
Comment (sgandon) [inline services/components-api-service-rest-all-components/pom.xml]: Why did you remove snappy here ?

Comment (RyanSkraba) [inline services/components-api-service-rest-all-components/pom.xml]: Hello!  I probably shouldn't have glued this fix here -- it turns out that the snappy shared library is unnecessary and was NOT causing problems.  The root cause was the missing `libc6-compat` library.  I have validated this.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/567/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/567/)
